### PR TITLE
NMS-14161:Event/Alarm advanced search not passing search terms

### PR DESCRIPTION
**Pull request by psuthar-opennms**
### All Contributors

* [ ] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): http://issues.opennms.org/browse/NMS-14161

### CODE REVIEW

**Review by cgorantla — COMMENTED** [review]
LGTM!  Can you add test case for this ?

File: opennms-webapp/src/main/java/org/opennms/web/filter/FilterUtil.java [hunk 1]
```diff
@@ -50,4 +51,14 @@ public static String toFilterURL(List<String> filters) {
         }
         return( buffer.toString() );
     }
+
+    public static String[] parse(String filterString) {
+        String[] filterParameter = (filterString == null) ? null :Arrays.stream(filterString.split("&amp;"))
+                .filter(fp -> fp.startsWith("filter="))
+                .map(fp -> fp.replace("filter=", ""))
+                .map(fp -> fp = URLDecoder.decode(fp))
```
Comment (cgorantla) [inline opennms-webapp/src/main/java/org/opennms/web/filter/FilterUtil.java:59]: decode(fp) seems to be deprecated
```suggestion
                .map(fp -> fp = URLDecoder.decode(fp, StandardCharsets.UTF_8.name()))
```

File: opennms-webapp/src/main/java/org/opennms/web/filter/FilterUtil.java [hunk 2]
```diff
@@ -50,4 +51,14 @@ public static String toFilterURL(List<String> filters) {
         }
         return( buffer.toString() );
     }
+
+    public static String[] parse(String filterString) {
```
Comment (cgorantla) [inline opennms-webapp/src/main/java/org/opennms/web/filter/FilterUtil.java:55]: Please add a test case where `filterString` can have multiple filter expressions

File: opennms-webapp/src/test/java/org/opennms/web/event/EventControllerIT.java [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+        alarm.setSeverityId(3);
+        alarm.setCounter(100);
+        alarm.setLastEvent(getOnmsEvents().get(0));
+        alarm.setLogMsg("test");
+        OnmsAlarm[] alarms = {alarm};
+        return alarms;
+    }
+
+
+    private Event mapOnmsEventToEvent(OnmsEvent onmsEvent){
```
Comment (cgorantla) [inline opennms-webapp/src/test/java/org/opennms/web/event/EventControllerIT.java:223]: Do you need this mapping? 
Can you directly create Event

File: opennms-webapp/src/test/java/org/opennms/web/event/EventControllerIT.java [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * The Test Class for GraphResultsController.
+ *
+ * @author Alejandro Galue <agalue@opennms.org>
+ */
```
[the remark this answers is not in the document]
Comment (psuthar-opennms) [reply opennms-webapp/src/test/java/org/opennms/web/event/EventControllerIT.java]: removed unnecessary comments

File: opennms-webapp/src/test/java/org/opennms/web/event/EventControllerIT.java [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+public class EventControllerIT  {
```
[the remark this answers is not in the document]
Comment (psuthar-opennms) [reply opennms-webapp/src/test/java/org/opennms/web/event/EventControllerIT.java]: Removed comments

File: opennms-webapp/src/test/java/org/opennms/web/event/EventControllerIT.java [hunk 6]
```diff
@@ -0,0 +1,259 @@
...
+        alarmFilterController = new AlarmFilterController();
+        alarmFilterController.setServletContext(new MockServletContext("file:src/main/webapp"));
+        alarmFilterController.setWebAlarmRepository(m_webAlarmRepository);
+        alarmFilterController.setFavoriteService(favoriteService);
+        alarmFilterController.afterPropertiesSet();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        //verifyNoMoreInteractions(m_webEventRepository);
```
[the remark this answers is not in the document]
Comment (psuthar-opennms) [reply opennms-webapp/src/test/java/org/opennms/web/event/EventControllerIT.java]: method removed

File: opennms-webapp/src/main/java/org/opennms/web/filter/FilterUtil.java (hunk 1 above)
Comment (psuthar-opennms) [reply opennms-webapp/src/main/java/org/opennms/web/filter/FilterUtil.java:59]: removed deprecated method

File: opennms-webapp/src/main/java/org/opennms/web/filter/FilterUtil.java (hunk 2 above)
Comment (psuthar-opennms) [reply opennms-webapp/src/main/java/org/opennms/web/filter/FilterUtil.java:55]: Converted test cases to use multiple filters

File: opennms-webapp/src/test/java/org/opennms/web/event/EventControllerIT.java (hunk 3 above)
Comment (psuthar-opennms) [reply opennms-webapp/src/test/java/org/opennms/web/event/EventControllerIT.java]: Created events directly now

**Comment by cgorantla** [comment]
@RangerRick Do we have any existing Utils to decode URLEncoded parameters ?

**Review by RangerRick — APPROVED** [review]
Other than that one question, it seems reasonable to me.

File: opennms-webapp/src/main/java/org/opennms/web/filter/FilterUtil.java [hunk 7]
```diff
@@ -50,4 +51,14 @@ public static String toFilterURL(List<String> filters) {
...
+        String[] filterParameter = (filterString == null) ? null :Arrays.stream(filterString.split("&amp;"))
```
[the remark this answers is not in the document]
Comment (psuthar-opennms) [reply opennms-webapp/src/main/java/org/opennms/web/filter/FilterUtil.java:56]: there is not specific reason for that but doing that first will improve performance?

[the remark this answers is not in the document]
Comment (RangerRick) [reply opennms-webapp/src/main/java/org/opennms/web/filter/FilterUtil.java:56]: I would expect so. Also just simplifies the code a bit...

[the remark this answers is not in the document]
Comment (psuthar-opennms) [reply opennms-webapp/src/main/java/org/opennms/web/filter/FilterUtil.java]: Changes done

**Comment by psuthar-opennms** [comment]
> @psuthar-opennms Looks like this is failing in smoke tests. Did you verify if it is working properly after these changes!

Changes done and now smoke test cases are passing

**Review by cgorantla — APPROVED** [review]
LGTM!

**Pull request merged**